pkg/util/util.go

```diff
@@ -0,0 +1,346 @@
+	if (err != nil) && apierrors.IsAlreadyExists(err) {
+		br, err = client.ObjectstorageV1alpha1().BucketRequests(br.Namespace).Get(ctx, br.Name, metav1.GetOptions{})
+	}
+	return br, err
+}
+
+func CreateBucketAccessRequest(ctx context.Context, client bucketclientset.Interface, bar *types.BucketAccessRequest) (*types.BucketAccessRequest, error) {
+	bar, err := client.ObjectstorageV1alpha1().BucketAccessRequests(bar.Namespace).Create(ctx, bar, metav1.CreateOptions{})
+	if (err != nil) && apierrors.IsAlreadyExists(err) {
+		bar, err = client.ObjectstorageV1alpha1().BucketAccessRequests(bar.Namespace).Get(ctx, bar.Name, metav1.GetOptions{})
+	}
+	return bar, err
+}
+
+func CreateBucketClass(ctx context.Context, client bucketclientset.Interface, bc *types.BucketClass) (*types.BucketClass, error) {
+	bc, err := client.ObjectstorageV1alpha1().BucketClasses().Create(ctx, bc, metav1.CreateOptions{})
+	if (err != nil) && apierrors.IsAlreadyExists(err) {
+		bc, err = client.ObjectstorageV1alpha1().BucketClasses().Get(ctx, bc.Name, metav1.GetOptions{})
+	}
+	return bc, err
+}
+
+func CreateBucketAccessClass(ctx context.Context, client bucketclientset.Interface, bac *types.BucketAccessClass) (*types.BucketAccessClass, error) {
+	bac, err := client.ObjectstorageV1alpha1().BucketAccessClasses().Create(ctx, bac, metav1.CreateOptions{})
+	if (err != nil) && apierrors.IsAlreadyExists(err) {
+		bac, err = client.ObjectstorageV1alpha1().BucketAccessClasses().Get(ctx, bac.Name, metav1.GetOptions{})
+	}
+	return bac, err
+}
+
+func CreateBucket(ctx context.Context, client bucketclientset.Interface, b *types.Bucket) (*types.Bucket, error) {
+	b, err := client.ObjectstorageV1alpha1().Buckets().Create(ctx, b, metav1.CreateOptions{})
+	if (err != nil) && apierrors.IsAlreadyExists(err) {
+		b, err = client.ObjectstorageV1alpha1().Buckets().Get(ctx, b.Name, metav1.GetOptions{})
+	}
+	return b, err
+}
+
+func CreateBucketAccess(ctx context.Context, client bucketclientset.Interface, ba *types.BucketAccess) (*types.BucketAccess, error) {
+	ba, err := client.ObjectstorageV1alpha1().BucketAccesses().Create(ctx, ba, metav1.CreateOptions{})
+	if (err != nil) && apierrors.IsAlreadyExists(err) {
+		ba, err = client.ObjectstorageV1alpha1().BucketAccesses().Get(ctx, ba.Name, metav1.GetOptions{})
+	}
+	return ba, err
+}
+
+func getCRDClient() (apiextensions.CustomResourceDefinitionInterface, error) {
+	config, err := func() (*rest.Config, error) {
+		kubeConfig := os.Getenv("KUBECONFIG")
+
+		if kubeConfig != "" {
+			return clientcmd.BuildConfigFromFlags("", kubeConfig)
+		}
+		return rest.InClusterConfig()
+	}()
+
+	crdClientset, err := apiextensions.NewForConfig(config)
+	if err != nil {
+		glog.Fatalf("could not initialize crd client: %v", err)
+		return nil, err
+	}
+	return crdClientset.CustomResourceDefinitions(), err
+}
+
+func RegisterCRDs(ctx context.Context, client apiextensions.CustomResourceDefinitionInterface) error {
```

Comment:
I'd argue these should be set up/managed in a test environment/cluster by the test automation, not by the test(s) themselves.

Reply:
I agree, will be reworked once we have a proper setup mechanism.